plugins/provisioners/ansible/plugin.rb

```diff
@@ -3,21 +3,73 @@
 module VagrantPlugins
   module Ansible
     class Plugin < Vagrant.plugin("2")
+
       name "ansible"
       description <<-DESC
-      Provides support for provisioning your virtual machines with
-      Ansible playbooks.
+      Provides support for provisioning your virtual machines with Ansible
+      from the Vagrant host (`ansible`) or from the guests (`ansible_local`).
       DESC
 
-      config(:ansible, :provisioner) do
-        require File.expand_path("../config", __FILE__)
-        Config
+      config("ansible", :provisioner) do
+        require_relative "config/host"
+        Config::Host
+      end
+
+      config("ansible_local", :provisioner) do
```

Comment:
The name `ansible_local` corresponds to [Packer's `ansible-local`](https://www.packer.io/docs/provisioners/ansible-local.html), but following underscore style used in Vagrant (e.g. `chef_solo`,...).

Note that this new provisioner is potentially able to also execute Ansible against remote machines (see the trick in `ansible_local.html.md`), and I personally would find a bit more correct to name it `ansible_on_guest` (or something similar). What do you think?